x-pack/plugins/observability/public/pages/rules/rules.tsx

```diff
@@ -141,13 +142,15 @@ export function RulesPage({ activeTab = RULES_TAB_NAME }: RulesPageProps) {
         <AddRuleFlyout
           consumer={ALERTS_FEATURE_ID}
           filteredRuleTypes={filteredRuleTypes}
+          validConsumers={observabilityRuleCreationValidConsumers}
           onClose={() => {
             setAddRuleFlyoutVisibility(false);
           }}
           onSave={() => {
             setRefresh(new Date());
             return Promise.resolve();
           }}
+          useRuleProducer={true}
```

Comment:
Is `useRuleProducer` about showing or not showing the drop-down? If yes, couldn't we infer it from the existence of `validConsumers`?

Reply:
`useRuleProducer` is not about showing the dropdown but to avoid to use the `alerts` as a consumer when you know that the rule has been created inside of o11y rule management page. Two different concept, it is just mean to use the producer of the rule type as a consumer